### PR TITLE
openapi: Align KubeVirt API with K8S

### PR DESCRIPTION
**Pull request by lukas-bednar**
* adding missing parameters for delete and list operation
* fixes Produces/Consumes definitions
* Align HTTP return codes with K8S

### CODE REVIEW

File: pkg/virt-api/rest/kubeproxy.go [hunk 1]
```diff
@@ -138,39 +142,41 @@ func GenericResourceProxy(ws *restful.WebService, ctx context.Context, gvr schem
...
-			Returns(http.StatusNotFound, "Not Found", nil), ws,
+			Returns(http.StatusUnauthorized, "Unauthorized", nil), ws,
 	))
 
 	ws.Route(
 		ws.PATCH(ResourcePath(gvr)).
-			Produces(mime.MIME_JSON_PATCH).
-			Operation("updateNamespaced"+objKind).
-			To(endpoints.MakeGoRestfulWrapper(patch)).Writes(objExample).
+			Consumes(mime.MIME_JSON_PATCH).
```
Comment (lukas-bednar) [inline pkg/virt-api/rest/kubeproxy.go]: Need to add `"application/merge-patch+json"`

**Comment by fabiand** [comment]
What's the status of this PR?

**Comment by lukas-bednar** [comment]
ok to test

**Comment by lukas-bednar** [comment]
@fabiand I was trying to get closer to k8s rest-api by adding list of query parameters because I need them to perform `watch` operation on elements with specific `label` like `filedselector`.
And it turned out that there are other things which could be added and fixed.
There were wrong return and consume types for some calls, for example `delete` or `watch`.
Then I tried `patch` operation which was failing on unsupported content-type.

I think this patch is ready to be tested, but apparently can not mark it as `ok to test` :-)

**Comment by rmohr** [comment]
ok to test

**Comment by fabiand** [comment]
Any idea why the migration tests are failing?

**Comment by rmohr** [comment]
@fabiand there might be the possibility that it is because of a libvirt issue which we saw when we added vnc. Let's see, if it happens again, after we  have merged #647.

File: pkg/virt-api/rest/kubeproxy.go [hunk 2]
```diff
@@ -138,39 +142,41 @@ func GenericResourceProxy(ws *restful.WebService, ctx context.Context, gvr schem
...
 	// TODO, implement watch. For now it is here to provide swagger doc only
 	ws.Route(addWatchGetListParams(
 		ws.GET("/watch/"+gvr.Resource).
 			Produces(mime.MIME_JSON).
 			Operation("watch"+objKind+"ListForAllNamespaces").
-			To(NotImplementedYet).Writes(listExample).
+			To(NotImplementedYet).Writes(metav1.WatchEvent{}).
 			Doc("Watch a "+objKind+"List object.").
-			Returns(http.StatusOK, "OK", listExample).
-			Returns(http.StatusNotFound, "Not Found", nil), ws,
```
Comment (rmohr) [inline pkg/virt-api/rest/kubeproxy.go]: Are you sure that the API can never return 404, or is that already globally added?

Comment (lukas-bednar) [reply pkg/virt-api/rest/kubeproxy.go]: I think API can return 404 for sure, but when I was listing k8s spec I can not see 404 be defined there ...
I don't have any reason to remove it except the fact that it is missing in k8s.
If you want I can put it back.

**Comment by lukas-bednar** [comment]
I rebased change on latest master.

**Pull request merged**